### PR TITLE
fix(make): use the default rust dir for build/build-op

### PR DESCRIPTION
**Pull request by jsvisa**
Currently, we use `cargo build --target $(shell rustc -Vv | grep host | cut -d ' ' -f2)` to run the `make build` process, this will put the result into the directory as below:

```
target/aarch64-apple-darwin/release/reth # for macOS
target/x86_64-unknown-linux-gnu/release/reth # for Linux
```

This is inconvenience to run reth command(eg: `./target/aarch64-apple-darwin/release/reth node --dev`), so propose to use the default rust target directory(target/{release,debug}) for the build process.

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
yeah no idea why we do this

am I missing anything @onbjerg @shekhirin ?

**Comment by onbjerg** [comment]
we use this for the cross-platform docker builds, so we cannot accommodate this

**Comment by mattsse** [comment]
maybe we could rename this to docker-build instead and add these changes as build?

I can see how make build would be inconvenient here

**Comment by jsvisa** [comment]
> we use this for the cross-platform docker builds which requires a very specific structure (target/$platform/$profile), so we cannot accommodate this

@onbjerg Seems docker build not using the `build` target:


https://github.com/paradigmxyz/reth/blob/8dbffee15f6b92db4a02d5ca70aa4864acfbc29f/Makefile#L218-L233

which uses `build-%` instead(not this pr related `build`)

https://github.com/paradigmxyz/reth/blob/8dbffee15f6b92db4a02d5ca70aa4864acfbc29f/Makefile#L102-L105

File: Makefile [hunk 1]
```diff
@@ -303,8 +303,7 @@ db-tools: ## Compile MDBX debugging tools.
 	@echo "Run \"$(DB_TOOLS_DIR)/mdbx_chk\" for the MDBX db file integrity check."
 
 .PHONY: update-book-cli
-update-book-cli: ## Update book cli documentation.
-	cargo build --bin reth --features "$(FEATURES)" --profile "$(PROFILE)"
+update-book-cli: build ## Update book cli documentation.
```
Comment (onbjerg) [inline Makefile:306]: this builds in release mode every time by default which is probably not ideal

**Review by onbjerg — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (jsvisa) [reply Makefile:306]: Yeah, and it depends on weather the binary reth changed or not, if nothing changed, I think cargo would not need to recompile the binary

Comment (emhane) [reply Makefile:306]: should we revert it @onbjerg ?

**Pull request merged**